### PR TITLE
fence_openstack: Major rework of the original OpenStack agent.

### PR DESCRIPTION
**Pull request by bdperkin**
* Remove unnecessary imports
* Change path append to macro
* Check for novaclient import separate from keystoneclient
* Ignore urllib3 warnings due to self-signed certificates
* Enable logging to syslog
* Add list and list-status functionality
* Deal with many more client exceptions
* Change from REBOOT_HARD to HARD due to API change
* Replace nova_run_command with get_power_status and set_power_status
* Added cacert option to authenticate against self-signed certificates
* Added apitimeout to pass to the client session

### CODE REVIEW

**Comment by fabbione** [comment]
ok to test

File: agents/openstack/fence_openstack.py [hunk 1]
```diff
@@ -2,161 +2,304 @@
...
     if "--uuid" in options:
         options["--plug"] = options["--uuid"]
         del options["--uuid"]
-    elif "--help" not in options and options["--action"] in ["off", "on", \
-         "reboot", "status", "validate-all"] and "--plug" not in options:
+    elif (
+        "--help" not in options
+        and options["--action"] in ["off", "on", "reboot", "status", "validate-all"]
+        and "--plug" not in options
+    ):
```
[the remark this answers is not in the document]
Comment (bdperkin) [reply agents/openstack/fence_openstack.py]: Fixed in: commit 52830d0a5b95fef91275fb6e5657f047b9477490

```
-    elif (
-        "--help" not in options
-        and options["--action"] in ["off", "on", "reboot", "status", "validate-all"]
-        and "--plug" not in options
-    ):
+    elif ("--help" not in options
+          and options["--action"] in ["off", "on", "reboot", "status", "validate-all"]
+          and "--plug" not in options):
```

File: agents/openstack/fence_openstack.py [hunk 2]
```diff
@@ -2,161 +2,304 @@
...
-        fail_usage("Failed: You have to enter plug number or machine identification", stop_after_error)
+        fail_usage(
+            "Failed: You have to enter plug number or machine identification",
+            stop_after_error,
+        )
 
     docs = {}
     docs["shortdesc"] = "Fence agent for OpenStack's Nova service"
-    docs["longdesc"] = "fence_openstack is a Fencing agent \
+    docs[
```
[the remark this answers is not in the document]
Comment (bdperkin) [reply agents/openstack/fence_openstack.py]: Fixed in: commit 52830d0a5b95fef91275fb6e5657f047b9477490

```
-    docs[
-        "longdesc"
-    ] = "fence_openstack is a Fencing agent \
+    docs["longdesc"] = "fence_openstack is a Fencing agent \
```

**Review by oalbrigt — APPROVED** [review]

**Comment by oalbrigt** [comment]
LGTM. Thanks.

**Pull request merged**